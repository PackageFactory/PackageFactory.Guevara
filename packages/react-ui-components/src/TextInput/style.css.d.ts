export interface IStyleCss {
  'textInput': string;
  'textInput--disabled': string;
}

export const locals: IStyleCss;
