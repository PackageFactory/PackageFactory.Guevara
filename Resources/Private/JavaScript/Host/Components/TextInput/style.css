.textInput {
    padding-bottom: $spacing;
}
.textInput__input {
    width: 100%;
    height: 40px;
    padding: 0 14px;
    border: 0;
    background: $neutral;

    &:focus {
        outline: 0;
        background: $brightest;
        color: $darkest;
    }
}
.textInput__input::-webkit-input-placeholder {
    font-family: 'Noto Sans', sans-serif;
    color: $bright;
}
.textInput--invalid {
    outline: 2px solid $error;
}
