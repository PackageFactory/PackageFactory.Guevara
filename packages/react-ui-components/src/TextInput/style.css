.textInput {
    composes: reset from './../reset.css';
    /* TODO: should eventually be moved to reset.css, but there maybe be plenty of side-effects */
    font-size: 14px;
    width: 100%;
    height: var(--goldenUnit);
    padding: 0 14px;
    border: 0;
    background: var(--brandColorsContrastNeutral);
    color: white;

    &:focus {
        outline: 0;
        background: var(--brandColorsContrastBrightest);
        color: var(--brandColorsContrastDarkest);
    }
}
.textInput::-webkit-input-placeholder {
    font-family: 'Open Sans', sans-serif;
    font-weight: normal;
    color: var(--brandColorsContrastBright);
}
.textInput--invalid {
    outline: 2px solid var(--brandColorsError) !important;
    color: var(--brandColorsError) !important;
}
.textInput--highlight {
    outline: 2px solid var(--brandColorsSuccess);
}

.textInput--disabled {
    opacity: .65;
    cursor: not-allowed;
}
