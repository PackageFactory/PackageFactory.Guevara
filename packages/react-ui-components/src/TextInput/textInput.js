import React, {PureComponent} from 'react';
import PropTypes from 'prop-types';
import mergeClassNames from 'classnames';
import omit from 'lodash.omit';

class TextInput extends PureComponent {
    static propTypes = {
        /**
         * An optional className to render on the textarea node.
         */
        className: PropTypes.string,

        /**
         * An optional className for the surrounding container div.
         */
        containerClassName: PropTypes.string,

        /**
         * An optional HTML5 placeholder.
         */
        placeholder: PropTypes.string,

        /**
         * The handler which will be called once the user changes the value of the input.
         */
        onChange: PropTypes.func,

        /**
         * The handler which will be called once the user takes focus on the input.
         */
        onFocus: PropTypes.func,

        /**
         * The handler which will be called once the user leaves focus of the input.
         */
        onBlur: PropTypes.func,

        /**
         * This prop controls if the CheckBox is disabled or not.
         */
        disabled: PropTypes.bool,

        /**
         * This prob controls what function is triggered when the enter key is pressed
         */
        onEnterKey: PropTypes.func,

        /**
         * A prop of which type this input field is eg password
         */
        type: PropTypes.string,

        /**
         * Set the focus to this input element after mount
         */
        setFocus: PropTypes.bool,

        /**
         * An optional css theme to be injected.
         */
        theme: PropTypes.shape({
            textInput: PropTypes.string
        }).isRequired
    };

    componentDidMount() {
        if (this.props.setFocus) {
            this.inputRef.focus();
        }
    }

    handleKeyPress = e => {
        const enterKeyCode = 13;
        const keyCode = e.keyCode || e.which;
        const {onEnterKey} = this.props;
        if (keyCode === enterKeyCode && typeof onEnterKey === 'function') {
            onEnterKey();
        }
    }

    handleValueChange = e => {
        const value = e.target.value;
        const {onChange} = this.props;

        if (onChange) {
            onChange(value);
        }
    }

    render() {
        const {
            placeholder,
            className,
            theme,
            containerClassName,
            disabled,
            type,
            ...restProps
        } = this.props;

        const rest = omit(restProps, ['onEnterKey', 'setFocus']);
        const classNames = mergeClassNames({
            [className]: className && className.length,
            [theme.textInput]: true,
            [theme['textInput--disabled']]: disabled
        });

        const inputRef = el => {
            this.inputRef = el;
        };

        return (
            <div className={containerClassName}>
                <input
                    {...rest}
                    className={classNames}
                    role="textbox"
                    aria-multiline="false"
                    aria-disabled={disabled ? 'true' : 'false'}
                    type={type}
                    placeholder={placeholder}
                    disabled={disabled}
                    onChange={this.handleValueChange}
                    onKeyPress={this.handleKeyPress}
                    ref={inputRef}
                    />
            </div>
        );
    }
}

export default TextInput;
