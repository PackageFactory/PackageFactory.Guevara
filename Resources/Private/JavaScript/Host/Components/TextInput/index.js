import React, {PropTypes} from 'react';
import uuid from 'uuid';
import mergeClassNames from 'classnames';
import {executeCallback} from 'Host/Abstracts/';
import Label from 'Host/Components/Label/';
import style from './style.css';

const onChangeHandler = (e, cb) => {
    const value = e.target.value.substr(0, 140);

    if (cb) {
        cb(value);
    }
};

const TextInput = props => {
    const {
        label,
        placeholder,
        className,
        isValid,
        onChange,
        onFocus,
        onBlur
    } = props;
    const classNames = mergeClassNames({
        [className]: className && className.length,
        [style.textInput]: true
    });
    const inputClassNames = mergeClassNames({
        [style.textInput__input]: true,
        [style['textInput--invalid']]: !isValid
    });
    const id = uuid.v1();

    return (
        <div className={classNames}>
            <Label htmlFor={id} label={label} />
            <input
                className={inputClassNames}
                id={id}
                type="text"
                placeholder={placeholder}
                onChange={e => onChangeHandler(e, onChange)}
                onFocus={() => executeCallback(onFocus)}
                onBlur={() => executeCallback(onBlur)}
                />
        </div>
    );
};
TextInput.propTypes = {
    label: PropTypes.string.isRequired,
    placeholder: PropTypes.string,

    // Style related propTypes.
    className: PropTypes.string,

    // State related propTypes.
    isValid: PropTypes.bool.isRequired,

    // Interaction related propTypes.
    onChange: PropTypes.func,
    onFocus: PropTypes.func,
    onBlur: PropTypes.func
};
TextInput.defaultProps = {
    isValid: true
};

export default TextInput;
