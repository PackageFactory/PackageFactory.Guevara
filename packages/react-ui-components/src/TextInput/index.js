import {themr} from 'react-css-themr';
import identifiers from './../identifiers.js';
import style from './style.css';
import TextInput from './textInput.js';
import withValidationResultRendering from './../_lib/withValidationResultRendering.js';

const ThemedTextInput = themr(identifiers.textInput, style)(TextInput);
const TextInputWithValidationResultRendering = withValidationResultRendering(ThemedTextInput);

export default TextInputWithValidationResultRendering;
