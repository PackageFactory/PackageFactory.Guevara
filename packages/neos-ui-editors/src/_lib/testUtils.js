import {PureComponent, Children} from 'react';
import PropTypes from 'prop-types';
import {SynchronousRegistry, SynchronousMetaRegistry} from '@neos-project/neos-ui-extensibility/src/registry';

const globalRegistry = new SynchronousMetaRegistry();

// i18n Registry
class FakeI18NRegistry extends SynchronousRegistry {
    translate(key) {
        return key;
    }
}
globalRegistry.set('i18n', new FakeI18NRegistry());

// Data Source DataLoader
export const MockDataSourceDataLoader = {
    _currentPromise: null,
    _currentPromiseResolveFn: null,

    // API in tests
    reset() {
        this._currentPromise = null;
    },

    resolveCurrentPromise(result) {
        this._currentPromiseResolveFn(result);
        return this._currentPromise;
    },

    // API of DataLoader
    resolveValue() {
        if (!this._currentPromise) {
            this._currentPromise = new Promise(resolve => {
                this._currentPromiseResolveFn = resolve;
            });
        }
        return this._currentPromise;
    }
};
globalRegistry.set('dataLoaders', new SynchronousRegistry());
globalRegistry.get('dataLoaders').set('DataSources', MockDataSourceDataLoader);

// wrapper
export class WrapWithMockGlobalRegistry extends PureComponent {
    static propTypes = {
        children: PropTypes.element.isRequired
    };

    static childContextTypes = {
        globalRegistry: PropTypes.object.isRequired,
        configuration: PropTypes.object.isRequired
    };

    getChildContext() {
        const configuration = {};
        return {globalRegistry, configuration};
    }

    render() {
        return Children.only(this.props.children);
    }
}
