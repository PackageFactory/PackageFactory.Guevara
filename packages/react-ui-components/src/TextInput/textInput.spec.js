import sinon from 'sinon';
import {createShallowRenderer} from './../_lib/testUtils.js';
import TextInput from './textInput.js';

const defaultProps = {
    theme: {}
};
const shallow = createShallowRenderer(TextInput, defaultProps);

test('should render an "input" node.', () => {
    const input = shallow().find('input');

    expect(input.type()).toBe('input');
});
test('should add the passed "className" prop to the rendered button if passed.', () => {
    const input = shallow({className: 'testClassName'}).find('input');

    expect(input.hasClass('testClassName')).toBeTruthy();
});
test('should call the passed "onChange" prop with the value of the input when changing it.', () => {
    const onChange = sinon.spy();
    const input = shallow({onChange}).find('input');

    input.simulate('change', {
        target: {
            value: 'my value'
        }
    });

    expect(onChange.callCount).toBe(1);
    expect(onChange.args[0][0]).toBe('my value');
});
test('should throw no error if no "onChange" prop was passed when changing the value of the input.', () => {
    const input = shallow().find('input');
    const fn = () => {
        input.simulate('change', {
            target: {
                value: 'my value'
            }
        });
    };

    expect(fn).not.toThrow();
});
