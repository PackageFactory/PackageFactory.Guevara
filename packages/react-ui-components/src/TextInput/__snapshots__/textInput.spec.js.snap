// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<TextInput/> should render correctly. 1`] = `
<div>
  <input
    TooltipComponent={[Function]}
    aria-disabled="false"
    aria-multiline="false"
    className=""
    onChange={[Function]}
    onKeyPress={[Function]}
    role="textbox"
  />
</div>
`;
